incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/config/MqttKafkaHeaderHelper.java

```diff
@@ -0,0 +1,169 @@
+
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+
+import io.aklivity.zilla.runtime.binding.mqtt.kafka.internal.types.KafkaHeaderFW;
+import io.aklivity.zilla.runtime.binding.mqtt.kafka.internal.types.OctetsFW;
+import io.aklivity.zilla.runtime.binding.mqtt.kafka.internal.types.String16FW;
+import io.aklivity.zilla.runtime.binding.mqtt.kafka.internal.types.stream.KafkaMergedDataExFW;
+
+public class MqttKafkaHeaderHelper
+{
+    public static final String KAFKA_TOPIC_HEADER_NAME = "zilla:topic";
+    public static final String KAFKA_LOCAL_HEADER_NAME = "zilla:local";
+    public static final String KAFKA_TIMEOUT_HEADER_NAME = "zilla:timeout-ms";
+    public static final String KAFKA_CONTENT_TYPE_HEADER_NAME = "zilla:content-type";
+    public static final String KAFKA_FORMAT_HEADER_NAME = "zilla:format";
+    public static final String KAFKA_REPLY_TO_HEADER_NAME = "zilla:reply-to";
+    public static final String KAFKA_CORRELATION_ID_HEADER_NAME = "zilla:correlation-id";
+    public final OctetsFW kafkaTopicHeaderOctets;
+    public final OctetsFW kafkaLocalHeaderOctets;
+    public final OctetsFW kafkaTimeoutHeaderOctets;
+    public final OctetsFW kafkaContentTypeHeaderOctets;
+    public final OctetsFW kafkaFormatHeaderOctets;
+    public final OctetsFW kafkaReplyToHeaderOctets;
+    public final OctetsFW kafkaCorrelationHeaderOctets;
+    private final Map<String, Consumer<OctetsFW>> visitors;
+    private final MutableDirectBuffer octetsBuffer;
+    private final OctetsFW emptyRO = new OctetsFW().wrap(new UnsafeBuffer(0L, 0), 0, 0);
+
+
+    public MqttKafkaHeaderHelper()
+    {
+        octetsBuffer = new UnsafeBuffer(new byte[8 * 1024]);
+        visitors = new HashMap<>();
+        visitors.put(KAFKA_TOPIC_HEADER_NAME, this::visitTopic);
+        visitors.put(KAFKA_LOCAL_HEADER_NAME, this::visitLocal);
+        visitors.put(KAFKA_TIMEOUT_HEADER_NAME, this::visitTimeout);
+        visitors.put(KAFKA_CONTENT_TYPE_HEADER_NAME, this::visitContentType);
+        visitors.put(KAFKA_FORMAT_HEADER_NAME, this::visitFormat);
+        visitors.put(KAFKA_REPLY_TO_HEADER_NAME, this::visitReplyTo);
+        visitors.put(KAFKA_CORRELATION_ID_HEADER_NAME, this::visitCorrelationId);
+
+        kafkaTopicHeaderOctets = stringToOctets(KAFKA_TOPIC_HEADER_NAME);
+        kafkaLocalHeaderOctets = stringToOctets(KAFKA_LOCAL_HEADER_NAME);
+        kafkaTimeoutHeaderOctets = stringToOctets(KAFKA_TIMEOUT_HEADER_NAME);
+        kafkaContentTypeHeaderOctets = stringToOctets(KAFKA_CONTENT_TYPE_HEADER_NAME);
+        kafkaFormatHeaderOctets = stringToOctets(KAFKA_FORMAT_HEADER_NAME);
+        kafkaReplyToHeaderOctets = stringToOctets(KAFKA_REPLY_TO_HEADER_NAME);
+        kafkaCorrelationHeaderOctets = stringToOctets(KAFKA_CORRELATION_ID_HEADER_NAME);
+    }
+
+    public List<String> topicNames;
+    public String local;
+    public int timeout;
+    public String contentType;
+    public String format;
+    public String replyTo;
```

Comment:
Suggest retaining contentType, format and replyTo as a separate `OctetsFW` flyweight, wrapped based on input header value offset and limit. This will prevent all the UTF-8 decoding that is happening now, also eliminates the UTF-8 encoding needed when writing them out.

Retaining `timeout` as `int` is all good.

Note that retaining a reference to the actual `OctetsFW` instance passed in to the visit method will not work as it is the "cursor" used by foreach and will change during iteration.

At the moment, we are retaining a direct reference to that for `correlationId` which only works if `zilla:correlation-id` is last and therefore the "cursor" flyweight is not further modified, but that is not the general case as headers can be in any order, so `zilla:correlation-id` could come up in the middle of the kafka headers.